wpiutil/src/main/native/include/wpi/protobuf/ProtobufCallbacks.h

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include "wpi/protobuf/Protobuf.h"
+#include "wpi/SmallVector.h"
+
+#include <span>
+
+#include "pb.h"
+
+#include <fmt/format.h>
+
+namespace wpi {
+
+using UnpackStringType = std::string;
+using UnpackBytesType = wpi::SmallVector<uint8_t, 128>;
+using StdVectorUnpackBytesType = std::vector<uint8_t>;
+
+template <class T>
+concept IntegerLike = std::is_integral_v<T>;
+
+template <class T>
+concept FloatingPointLike = std::is_floating_point_v<T>;
```

Comment:
The standard library already has [integral](https://en.cppreference.com/w/cpp/concepts/integral) and [floating_point](https://en.cppreference.com/w/cpp/concepts/floating_point) concepts.